spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/AsyncRabbitTemplate.java

```diff
@@ -0,0 +1,583 @@
+		}
+	}
+
+	/**
+	 * @see #isAutoStartup()
+	 * @param autoStartup true for auto start.
+	 */
+	public void setAutoStartup(boolean autoStartup) {
+		this.autoStartup = autoStartup;
+	}
+
+	/**
+	 * @see #getPhase()
+	 * @param phase the phase.
+	 */
+	public void setPhase(int phase) {
+		this.phase = phase;
+	}
+
+	/**
+	 * Set to true to enable the receipt of returned messages that cannot be delivered
+	 * in the form of a {@link AmqpMessageReturnedException}.
+	 * @param mandatory true to enable returns.
+	 */
+	public void setMandatory(boolean mandatory) {
+		if (mandatory) {
+			this.template.setReturnCallback(this);
+		}
+		this.template.setMandatory(mandatory);
+	}
+
+	/**
+	 * Set to true to enable publisher confirms. When enabled, the {@link RabbitFuture}
+	 * returned by the send and receive operation will have a
+	 * {@code ListenableFuture<Boolean>} in its {@code confirm} property.
+	 * @param enableConfirms true to enable publisher confirms.
+	 */
+	public void setEnableConfirms(boolean enableConfirms) {
+		this.enableConfirms = enableConfirms;
+		if (enableConfirms) {
+			this.template.setConfirmCallback(this);
+		}
+	}
+
+	/**
+	 * Send a message to the default exchange with the default routing key. If the message
+	 * contains a correlationId property, it must be unique.
+	 * @param message the message.
+	 * @return the {@link RabbitMessageFuture}.
+	 */
+	public RabbitMessageFuture sendAndReceive(Message message) {
+		return sendAndReceive(this.template.getExchange(), this.template.getRoutingKey(), message);
+	}
+
+	/**
+	 * Set the charset to be used when converting byte[] to/from String for
+	 * correlation Ids. Default: UTF-8.
+	 * @param charset the charset.
+	 */
+	public void setCharset(Charset charset) {
+		this.charset = charset;
+	}
+
+	/**
+	 * Send a message to the default exchange with the supplied routing key. If the message
+	 * contains a correlationId property, it must be unique.
+	 * @param routingKey the routing key.
+	 * @param message the message.
+	 * @return the {@link RabbitMessageFuture}.
+	 */
+	public RabbitMessageFuture sendAndReceive(String routingKey, Message message) {
+		return sendAndReceive(this.template.getExchange(), routingKey, message);
+	}
+
+	/**
+	 * Convert the object to a message and send it to the default exchange with the
+	 * default routing key.
+	 * @param message the message.
+	 * @return the {@link RabbitConverterFuture}.
+	 */
+	public RabbitConverterFuture convertSendAndReceive(Object message) {
+		return convertSendAndReceive(this.template.getExchange(), this.template.getRoutingKey(), message, null);
+	}
+
+	/**
+	 * Convert the object to a message and send it to the default exchange with the
+	 * provided routing key.
+	 * @param routingKey the routing key.
+	 * @param message the message.
+	 * @return the {@link RabbitConverterFuture}.
+	 */
+	public RabbitConverterFuture convertSendAndReceive(String routingKey, Object message) throws AmqpException {
```

Comment:
`AmqpException extends RuntimeException`, so no need to expose it in the `throws`.
Would you mind to explain the approach to do that or not for `RuntimeException`s?

I'm asking because the previous method doesn't do that.

Thanks.